src/python_testing/TC_MWOM_1_2.py

```diff
@@ -0,0 +1,122 @@
+            supported_modes = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes))
+
+            asserts.assert_greater_equal(len(supported_modes), 2, "SupportedModes must have at least 2 entries!")
+            asserts.assert_less_equal(len(supported_modes), 255, "SupportedModes must have at most 255 entries!")
+
+            modes = []
+            for m in supported_modes:
+                if m.mode in modes:
+                    asserts.fail("SupportedModes must have unique mode values!")
+                else:
+                    modes.append(m.mode)
+
+            labels = []
+            for m in supported_modes:
+                if m.label in labels:
+                    asserts.fail("SupportedModes must have unique mode label values!")
+                else:
+                    labels.append(m.label)
+
+            # common mode tags
+            commonTags = {0x0: 'Auto',
+                          0x1: 'Quick',
+                          0x2: 'Quiet',
+                          0x3: 'LowNoise',
+                          0x4: 'LowEnergy',
+                          0x5: 'Vacation',
+                          0x6: 'Min',
+                          0x7: 'Max',
+                          0x8: 'Night',
+                          0x9: 'Day'}
+
+            # derived cluster defined tags
+            derivedTags = [tag.value for tag in Clusters.MicrowaveOvenMode.Enums.ModeTag
+                       if tag is not Clusters.MicrowaveOvenMode.Enums.ModeTag.kUnknownEnumValue]
+
+            for m in supported_modes:
+                for t in m.modeTags:
+                    is_mfg = (0x8000 <= t.value and t.value <= 0xBFFF)
+                    asserts.assert_true(t.value in commonTags.keys() or t.value in derivedTags or is_mfg,
+                                        "Found a SupportedModes entry with invalid mode tag value!")
+                    if t.value == Clusters.MicrowaveOvenMode.Enums.ModeTag.kNormal:
+                        normal_present = True
+                    if t.value == Clusters.MicrowaveOvenMode.Enums.ModeTag.kDefrost:
+                        defrost_present = True
+            asserts.assert_true(normal_present, "The Supported Modes does not have an entry of Normal(0x4000)")
+
+        if self.check_pics("MWOM.S.A0001"):
+            self.print_step(3, "Read CurrentMode attribute")
+            current_mode = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.CurrentMode)
+
+            logging.info("CurrentMode: %s" % (current_mode))
+            asserts.assert_true(current_mode in modes, "CurrentMode is not a supported mode!")
+
+        if self.check_pics("MWOM.S.A0003"):
+            self.print_step(4, "Read OnMode attribute")
+            on_mode = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.OnMode)
```

Comment:
If OnMode is not allowed to be supported, wouldn't the "expect_success" part of this cause issues for devices where it is actually not implemented? 🤔

Reply:
Moved these tests to the YAML.